.github/workflows/e2e-test.yml

```diff
@@ -144,9 +145,27 @@ jobs:
       - name: Prepare test environment
         run: npm run test:e2e-setup
 
-      # Run Tests
-      - name: Run E2E Tests
-        run: npm run test:e2e
+      - name: First Run E2E Tests
+        id: first_run_e2e_tests
+        # Use +e to trap errors when running E2E tests.
+        shell: /bin/bash +e {0}
+        run: | 
+          npm run test:e2e -- --json --outputFile="$E2E_RESULT_FILEPATH"
+          
+          E2E_NUM_FAILED_TEST_SUITES=$(cat "$E2E_RESULT_FILEPATH" | jq '.numFailedTestSuites')
+
+          echo "::set-output name=FIRST_RUN_FAILED_TEST_SUITES::$(echo $E2E_NUM_FAILED_TEST_SUITES)"
+
+          if [[ ${E2E_NUM_FAILED_TEST_SUITES} -gt 0 ]]; then
+            echo '::notice::Some E2E tests failed in the first run but we will try them again in the second run.'
+            exit 0
+          fi
+      
+      - name: Re-try Failed E2E Files
+        if: ${{ steps.first_run_e2e_tests.outputs.FIRST_RUN_FAILED_TEST_SUITES > 0  }}
```

Comment:
Do we need to add a step to archive the generated output file or is that handled by GH actions?

Reply:
Addressed https://github.com/Automattic/woocommerce-payments/pull/4549/commits/c1941ea604691e0a337be16859ed19a35ac317f2